qiskit_algorithms/minimum_eigensolvers/qaoa.py

```diff
@@ -117,13 +125,19 @@ def __init__(
             callback: A callback that can access the intermediate data at each optimization step.
                 These data are: the evaluation count, the optimizer parameters for the ansatz, the
                 evaluated value, the metadata dictionary.
+            transpiler: An optional object with a `run` method allowing to transpile the circuits
+                that are produced within this algorithm. If set to `None`, these won't be transpiled.
+            transpiler_options: A dictionary of options to be passed to the transpiler's `run`
+                method as keyword arguments.
         """
         validate_min("reps", reps, 1)
 
         self.reps = reps
         self.mixer = mixer
         self.initial_state = initial_state
         self._cost_operator = None
+        self._transpiler = transpiler
+        self._transpiler_options = transpiler_options if transpiler_options is not None else {}
```

Comment:
Ah I see here that there is special treatment for None where by we create an empty dict that ** will be ok on (see an earlier item comment around this in the ampitude estimators)

Reply:
Yes, I try to avoid defining an empty `dict` as a default value to avoid any nasty side-effect!